packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -1835,7 +1830,9 @@ class AbciApp(
     final_states: Set[AppState] = set()
     event_to_timeout: EventToTimeout = {}
     cross_period_persisted_keys: List[str] = []
-
+    termination_round_cls: Optional[AppState] = None
+    termination_transition_function: Optional[AbciAppTransitionFunction] = None
+    termination_event: Optional[EventType] = None
```

Comment:
somethink like:
```
AnyAbciApp = with_termination(
    chain(
        (
            RegistrationAbci.AgentRegistrationAbciApp,
            SafeDeploymentAbci.SafeDeploymentAbciApp,
            TransactionSubmissionAbci.TransactionSubmissionAbciApp,
            ResetAndPauseAbci.ResetPauseABCIApp,
        ),
        abci_app_transition_mapping,
    )
)
```

Reply:
I like the idea of applying it at the composition level. That's what I was hoping you'd do. I suggest a slightly different syntax:

```
ChainedAbciApp = chain().add_termination(abci_app: Type[AbciApp] = TerminationAbciApp, transition_function: Type[AbciAppTransitionFunction] = ..)
```
where `ChainedTerminationAbciApp` can be e.g. the TermiationAbci + TransactionSettlementAbci chained.

Then this can be fined for instance here: https://github.com/valory-xyz/price-oracle/blob/main/packages/valory/skills/oracle_abci/composition.py#L73-L83